udata_hydra/utils/csv.py

```diff
@@ -2,10 +2,17 @@
 
 
 async def detect_csv_from_headers(check) -> bool:
```

Comment:
Can't we compute the `is_csvgz` in this method? And return `is_csv or is_csvgz` independently?
Same for `download_resource` returning an additional `is_csv`, not sure it is necessary.
I think we could keep the complexity inside the functions only if possible